osquery/events/linux/auditdnetlink.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <map>
+#include <memory>
+#include <set>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#include <libaudit.h>
+
+namespace osquery {
+
+/// Netlink status, used by AuditNetlink::acquireHandle()
+enum class NetlinkStatus { ActiveMutable, ActiveImmutable, Disabled, Error };
+
+/// Subscription handle to be used with AuditNetlink::getEvents()
+using NetlinkSubscriptionHandle = std::uint32_t;
+
+/// A single, prepared audit event record.
+struct AuditEventRecord final {
+  /// Record type (i.e.: AUDIT_SYSCALL, AUDIT_PATH, ...)
+  int type;
+
+  /// Event time.
+  unsigned long int time;
+
+  /// Audit event id that owns this record. Remember: PRIMARY KEY(id, timestamp)
+  std::string audit_id;
+
+  /// The field list for this record.
+  std::map<std::string, std::string> fields;
+};
+
+/// The subscriber context used by the AuditNetlink class to store the audit
+/// event records.
+struct AuditNetlinkSubscriberContext final {
+  /// This queue contains unprocessed events, waiting for a
+  /// AuditNetlink::getEvents() call to finalize them.
+  std::vector<AuditEventRecord> queue;
+
+  /// Queue mutex.
+  std::mutex queue_mutex;
+};
+
+class AuditdNetlink final {
+  /// This is the set of rules we have applied when configuring the service.
```

Comment:
Mind placing the private members at the bottom, and prefixing the list with `private:`?